### PR TITLE
Fix reading github context in detect-workflow

### PR DESCRIPTION
**Pull request by ianlewis**
Fix the detect-workflow action to get the github context from environment variables. The `github` context is not available in `actions.yml` so we can't use it as we have with reusable workflows.

We should consider using go-githubactions (#153) to get and manage the context as it will work within both actions and reusable workflows.

### CODE REVIEW

File: .github/actions/detect-workflow/main.go [hunk 1]
```diff
@@ -16,23 +16,57 @@ package main
...
 )
 
-// getStr returns a string value from the given Event map. Values are specified
+type action struct {
+	getenv func(string) string
+	event  map[string]any
+	client *github.OIDCClient
+}
+
+// TODO(github.com/slsa-framework/slsa-github-generator/issues/153): use go-githubactions
```
Comment (laurentsimon) [inline .github/actions/detect-workflow/main.go:36]: why do we need an external library for this?

**Review by laurentsimon — APPROVED** [review]

**Comment by laurentsimon** [comment]
Note: instead of using the env variables, you could also define a new Action argument that defaults to the github context (and keep the code as-is). We do that for the Scorecard Action to pass the token https://github.com/ossf/scorecard-action/blob/main/action.yaml#L30-L33

**Review by laurentsimon — COMMENTED** [review]
See my comment

File: .github/actions/detect-workflow/main.go (hunk 1 above)
Comment (ianlewis) [reply .github/actions/detect-workflow/main.go:36]: I'll leave out the TODO but we can discuss on #153

**Comment by ianlewis** [comment]
> Note: instead of using the env variables, you could also define a new Action argument that defaults to the github context (and keep the code as-is). We do that for the Scorecard Action to pass the token https://github.com/ossf/scorecard-action/blob/main/action.yaml#L30-L33

Hmm, interesting. When I tried to use the `github` context in `action.yaml` it gave me errors saying it was unrecognized and I couldn't find any info that it's available in `action.yaml`. Maybe it's only available in `inputs` and not in `env`? I'll give in another try today.

**Pull request merged**